pkg/authentication/authentication.go

```diff
@@ -0,0 +1,654 @@
+package authentication
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/github"
+	log "github.com/sirupsen/logrus"
+	"github.com/superplanehq/superplane/pkg/jwt"
+	"github.com/superplanehq/superplane/pkg/models"
+)
+
+type AuthenticationHandler struct {
+	jwtSigner *jwt.Signer
+}
+
+type AuthenticationUser struct {
```

Comment:
No need to prefix the struct with `Authentication`, since the package is already called `authentication`, so having `authentication.User` and `authentication.Handler` is probably better than `authentication.AuthenticationHandler` and authentication.AuthenticationUser